### ISSUE TITLE
An compile error occured when create register instance repeatly

### ISSUE THREAD

**Issue by mirages** [TypeStrong/ts-node#1168]
### Expected Behavior
No error occured, and use the new register instance compile the `.ts` file once when create a new register instance to cover an old one, the old one may be created by third-party library(e.g. `karma cli`).

The reason for create a new register is that, I want to use a special `RegisterOptions` for `.test.ts` files.

### Actual Behavior
By debugging the source code, I found that if create two register, the `require.extensions[.ts]` handler was nested registration by [`registerExtensions function`](https://github.com/TypeStrong/ts-node/blob/c11aa8ae59259d5b0ad17fe4b342e2b8782cea32/src/index.ts#L409) twice. And the `.ts` file was compiled twice, the old register's output as the new register's input, then some error occured, like below:
```node
TSError: ⨯ Unable to compile TypeScript:
error TS7022: '__awaiter' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.
2 var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
......
```

### Steps to reproduce the problem
1. First, create a `test.ts` for test the problem.
2. Then, create a `index.js`, and create two register instances directly in order to simplify the problem, then `require` the `test.ts` file.
3. Finally, run the `node index.js` command.

### Minimal reproduction
[ts-node-test](https://github.com/mirages/ts-node-test)

### Specifications

* ts-node version: 9.0.0
* node version: 13.9.0
* TypeScript version: 4.0.5
* Operating system and version: win7-64bit

**Comment by blakeembrey** [comment]
This is currently the expected behavior. It could be changed but there's always a possibility of double registration (e.g. if you had two `ts-node` instances we can't de-duplicate between them). A better solution is that you can ignore or only include certain paths with each instance registered.

**Comment by mirages** [comment]
@blakeembrey What's the meaning of " ignore or only include certain paths with each instance registered"? Could you give me more details or any demo?

**Comment by blakeembrey** [comment]
There is documentation in the README here: https://github.com/TypeStrong/ts-node#cli-and-programmatic-options. It looks like there's a typo but `scope` and `dir` would work. I can make a PR to your test repo if that helps further.

**Comment by cspotcode** [comment]
The options' names have since changed, but `scope` and `scopeDir` are more clearly implemented and documented by #1367
